src/index.ts

```diff
@@ -1,92 +1,166 @@
         this._name = name;
     }
+
+    addLevel(level: number): void {
+        this._levels.push(level);
+    }
+
+    removeLevel(level: number): void {
+        this._levels = this._levels.filter((element: number): boolean => element !== level);
+    }
+
+    get levels(): number[] {
+        return this._levels;
+    }
+
+    get name(): string {
+        return this._name;
+    }
 }
 
 class Level {
     // implement getters for fields and 'add/remove group' methods
+    _groups: string[] = [];
+    _name: string;
+    _description: string;
 
-    _groups;
-    _name;
-
-    constructor(name, description) {
-        this.name = name;
+    constructor(name: string, description: string) {
+        this._name = name;
         this._description = description;
     }
+
+    addGroup(group: string): void {
+        this._groups.push(group);
+    }
+
+    removeGroup(group: string) {
+        this._groups = this._groups.filter((element: string): boolean => element !== group);
+    }
+
+    get groups(): string[] {
+        return this._groups;
+    }
+
+    get name(): string {
+        return this._name;
+    }
+
+    get description(): string {
+        return this._description;
+    }
 }
 
 class Group {
     // implement getters for fields and 'add/remove student' and 'set status' methods
 
-    _area;
-    _status;
-    _students = []; // Modify the array so that it has a valid toSorted method*
+    _area: string = '';
+    _status: string = '';
+    _students: string[] = []; // Modify the array so that it has a valid toSorted method*
+    _directionName: string;
+    _levelName: string;
 
-    constructor(directionName, levelName) {
-        this.directionName = directionName;
-        this.levelName = levelName;
+    constructor(directionName: string, levelName: string) {
+        this._directionName = directionName;
+        this._levelName = levelName;
     }
 
-    showPerformance() {
-        const sortedStudents = this._students.toSorted((a, b) => b.getPerformanceRating() - a.getPerformanceRating());
+    showPerformance(): string[] {
+        const sortedStudents: string[] = this._students.toSorted((a: string, b: string) => b.getPerformanceRating() - a.getPerformanceRating());
         return sortedStudents;
     }
+
+    addStudent(student: string): void {
+        this._students.push(student);
+    }
+
+    removeStudent(student: string): void {
+        this._students = this._students.filter((element: string): boolean => element !== student);
+    }
+
+    set status(status: string) {
+        this._status = this._status + ' ' + status;
+    }
 }
 
 class Student {
     // implement 'set grade' and 'set visit' methods
 
-    _firstName;
-    _lastName;
-    _birthYear;
-    _grades = []; // workName: mark
-    _visits = []; // lesson: present
+    _firstName: string;
+    _lastName: string;
+    _birthYear: number;
+    _grades: number[] = []; // workName: mark
+    _visits: number[] = []; // lesson: present
```

Comment:
The same as `_lecturers `. In the comment, you see the sample of object structure.